checkQC/app.py

```diff
@@ -92,14 +94,19 @@ def run(self):
         log.info("Starting checkQC ({})".format(checkqc_version))
         log.info("------------------------")
         log.info("Runfolder is: {}".format(self._runfolder))
-        reports = self.configure_and_run()
-        if self.exit_status == 0:
-            log.info("Finished without finding any fatal qc errors.")
-        else:
-            log.error("Finished with fatal qc errors and will exit with non-zero exit status.")
-
-        if self._json_mode:
-            print(json.dumps(reports))
+        try:
+            reports = self.configure_and_run()
+            if self.exit_status == 0:
+                log.info("Finished without finding any fatal qc errors.")
+            else:
+                log.error("Finished with fatal qc errors and will exit with non-zero exit status.")
+
+            if self._json_mode:
+                print(json.dumps(reports))
+
+        except CheckQCException as e:
```

Comment:
Do we know of any more specific errors that tend to occur here, that we may want to catch? Fine as is though.

Reply:
In this case we want to catch all "handled" errors since we just log them to the console any way for the command line app.